src/main/java/cms/gov/madie/measure/services/GroupService.java

```diff
@@ -308,9 +309,64 @@ private TestCaseStratificationValue updateTestCaseStratification(
               .build();
     }
     testCaseStrata.setName(strataName);
+    handlePopulationChange(testCaseStrata, testCaseGroup);
+
     return testCaseStrata;
   }
 
+  private void handlePopulationChange(
+      TestCaseStratificationValue testCaseStrata, TestCaseGroupPopulation testCaseGroup) {
+    List<TestCasePopulationValue> testCasePopulationValues = testCaseStrata.getPopulationValues();
+    List<TestCasePopulationValue> testCasePopulationValuesFromGroup =
+        testCaseGroup.getPopulationValues();
+
+    if (!CollectionUtils.isEmpty(testCasePopulationValuesFromGroup)) {
+      if (!CollectionUtils.isEmpty(testCasePopulationValues)) {
+        // when there is added group population
+        if (testCasePopulationValuesFromGroup.size() > testCasePopulationValues.size()) {
+          for (int i = 0; i < testCasePopulationValuesFromGroup.size(); i++) {
+            TestCasePopulationValue testCasePopulationValueFromGroup =
+                testCasePopulationValuesFromGroup.get(i);
+            if (!findExistsInTestCasePopulationValues(
+                testCasePopulationValueFromGroup.getId(), testCasePopulationValues)) {
+              testCasePopulationValues.add(testCasePopulationValueFromGroup);
+            }
+          }
+        } // when there is deleted group population
+        else {
+          List<TestCasePopulationValue> tempTestCasePopulationValues = new ArrayList<>();
+          for (int i = 0; i < testCasePopulationValues.size(); i++) {
+            if (findExistsInTestCasePopulationValues(
+                testCasePopulationValues.get(i).getId(), testCasePopulationValuesFromGroup)) {
+              tempTestCasePopulationValues.add(testCasePopulationValues.get(i));
+            }
+          }
+          testCaseStrata.setPopulationValues(tempTestCasePopulationValues);
+        }
+
+      } // when there is new strat
+      else {
+        List<TestCasePopulationValue> tempTestCasePopulationValues = new ArrayList<>();
+        for (int i = 0; i < testCasePopulationValuesFromGroup.size(); i++) {
+          tempTestCasePopulationValues.add(testCasePopulationValuesFromGroup.get(i));
+        }
+        testCaseStrata.setPopulationValues(tempTestCasePopulationValues);
+      }
+    }
+  }
+
+  private boolean findExistsInTestCasePopulationValues(
+      String id, List<TestCasePopulationValue> testCasePopulationValues) {
+    boolean found = false;
+    for (int i = 0; i < testCasePopulationValues.size(); i++) {
```

Comment:
couldn't we simplify this function like this:
`
for (TestCasePopulationValue testCasePopulationValue : testCasePopulationValues) {
    if (id.equalsIgnoreCase(testCasePopulationValue.getId())) {
        return true;
    }
}
return false;
`

Reply:
I say, we use streams 
return testCasePopulationValues.stream().anyMatch(testCasePopulationValue -> id.equalsIgnoreCase(testCasePopulationValue.getId()));

Reply:
@RohitKandimalla Done, thanks Rohit!